p2panda-rs/src/encoder.rs

```diff
@@ -0,0 +1,282 @@
+use std::convert::{TryFrom, TryInto};
+
+use bamboo_rs_core::entry::{decode, MAX_ENTRY_SIZE};
+use bamboo_rs_core::{Entry as BambooEntry, Signature as BambooSignature, YamfHash};
+
+use crate::atomic::error::EntrySignedError;
+use crate::atomic::Blake2BArrayVec;
+use crate::atomic::{Entry, EntrySigned, Hash, LogId, Message, MessageEncoded, SeqNum};
+use crate::key_pair::KeyPair;
+use arrayvec::ArrayVec;
+use ed25519_dalek::PublicKey;
```

Comment:
Small style nitpick: These imports are external and would belong to the group above the internal imports